src/main/java/dao/BookRepositoryImpl.java

```diff
@@ -1,43 +1,53 @@
 package dao;
 
 import exception.DataProcessingException;
-import java.awt.print.Book;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
 import java.util.List;
-import org.hibernate.Session;
-import org.hibernate.SessionFactory;
-import org.hibernate.Transaction;
+import java.util.Optional;
+import model.Book;
+import org.springframework.stereotype.Repository;
 
+@Repository
 public class BookRepositoryImpl implements BookRepository {
-    private SessionFactory sessionFactory;
+    private final EntityManagerFactory entityManagerFactory;
+
+    public BookRepositoryImpl(EntityManagerFactory entityManagerFactory) {
+        this.entityManagerFactory = entityManagerFactory;
+    }
 
     @Override
     public Book save(Book book) {
-        Session session = null;
-        Transaction transaction = null;
-        try {
-            session = sessionFactory.openSession();
-            transaction = session.beginTransaction();
-            session.persist(book);
+        EntityTransaction transaction = null;
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            transaction = entityManager.getTransaction();
+            entityManager.persist(book);
             transaction.commit();
         } catch (Exception e) {
             if (transaction != null) {
                 transaction.rollback();
             }
-            throw new DataProcessingException("Can't save book to DB", e);
-        } finally {
-            if (session != null) {
-                session.close();
-            }
+            throw new DataProcessingException("Can't save book to DB");
         }
         return book;
     }
 
     @Override
     public List<Book> findAll() {
-        try (Session session = sessionFactory.openSession()) {
-            return session.createQuery("from Book", Book.class).getResultList();
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            return entityManager.createQuery("from Book", Book.class).getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get books from DB");
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            return Optional.ofNullable(entityManager.find(Book.class, id));
         } catch (Exception e) {
-            throw new DataProcessingException("Can't get books from DB", e);
+            throw new DataProcessingException("Can't find book with id " + id);

```

Comment:
since it is a find operation you should use `EntityNotFoundException`
```suggestion
            throw new EntityNotFoundException("Can't find book with id " + id, e);

```

Reply:
you and Liliia ask me to change, then change back...